soc/arm/nordic_nrf/common/soc_secure.c

```diff
@@ -9,23 +9,28 @@
 
 #include "nrf.h"
 
+#if CONFIG_TFM_PARTITION_PLATFORM
 #include "tfm_platform_api.h"
 #include "tfm_ioctl_api.h"
+#endif
 
 #if NRF_GPIO_HAS_SEL
 void soc_secure_gpio_pin_mcu_select(uint32_t pin_number, nrf_gpio_pin_sel_t mcu)
 {
+#if CONFIG_TFM_PARTITION_PLATFORM
 	uint32_t result;
 	enum tfm_platform_err_t err;
 
 	err = tfm_platform_gpio_pin_mcu_select(pin_number, mcu, &result);
 	__ASSERT(err == TFM_PLATFORM_ERR_SUCCESS, "TFM platform error (%d)", err);
 	__ASSERT(result == 0, "GPIO service error (%d)", result);
+#endif
 }
```

Comment:
Some static analyzers / compilers complain about unused variables.

Shall we add some ARG_UNUSED() macro's here ?

Reply:
N/A if we use a dedicated file. See other feedback.

Reply:
Addressed by excluding these functions from the build
There should be proper Kconfig dependencies to produce an error at config time if attempting to setup an invalid configuration.